docs/source/conf.py

```diff
@@ -101,6 +101,41 @@
     new_main_rst.write("".join(main_rst_content[2:]))
     new_main_rst.close()
 
+# Generate optional-plugins
+optional_plugins_path = os.path.join(root_path, "optional_plugins")
+api_optional_plugins_path = os.path.join(base_api_output_dir,
+                                         "optional-plugins")
+if not os.path.exists(api_optional_plugins_path):
+    os.makedirs(api_optional_plugins_path)
+with open(os.path.join(api_optional_plugins_path, "index.rst"),
+          'w') as optional_plugins_toc:
+    optional_plugins_toc.write(""".. index file for optional plugins API
+
+====================
+Optional Plugins API
+====================
+
+The following pages are auto-generated API documentation of optional
+Avocado plugins. This is not public API, it represents only current
+version and can be changed any time.
+
+.. toctree::
+   :maxdepth: 1
+
+    """)
+    for path in os.walk(optional_plugins_path).next()[1]:
+        name = "avocado_%s" % os.path.basename(path)
+        path = os.path.join(optional_plugins_path, path, name)
+        if not os.path.exists(path):
+            continue
+        output_dir = os.path.join(api_optional_plugins_path, name)
+        params = {"api_source_dir": path, "output_dir": output_dir,
+                  "exclude_dirs": ""}
+        process.run(apidoc_template % params)
+        # Remove the unnecessary generated files
+        os.unlink(os.path.join(output_dir, "modules.rst"))
+        optional_plugins_toc.write("\n   %s" % os.path.join(name, name))
```

Comment:
This brings failures in my local box when building the docs:

```
/home/cleber/src/avocado/avocado/docs/source/api/optional-plugins/avocado_resultsdb/avocado_resultsdb.rst:7: WARNING: autodoc: failed to import module u'avocado_resultsdb'; the following exception was raised:
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/sphinx/ext/autodoc.py", line 551, in import_object
    __import__(self.modname)
ImportError: No module named avocado_resultsdb
/home/cleber/src/avocado/avocado/docs/source/api/optional-plugins/avocado_robot/avocado_robot.rst:7: WARNING: autodoc: failed to import module u'avocado_robot'; the following exception was raised:
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/sphinx/ext/autodoc.py", line 551, in import_object
    __import__(self.modname)
ImportError: No module named avocado_robot
/home/cleber/src/avocado/avocado/docs/source/api/optional-plugins/avocado_runner_docker/avocado_runner_docker.rst:7: WARNING: autodoc: failed to import module u'avocado_runner_docker'; the following exception was raised:
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/sphinx/ext/autodoc.py", line 551, in import_object
    __import__(self.modname)
ImportError: No module named avocado_runner_docker
/home/cleber/src/avocado/avocado/docs/source/api/optional-plugins/avocado_runner_remote/avocado_runner_remote.rst:7: WARNING: autodoc: failed to import module u'avocado_runner_remote'; the following exception was raised:
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/sphinx/ext/autodoc.py", line 551, in import_object
    __import__(self.modname)
ImportError: No module named avocado_runner_remote
/home/cleber/src/avocado/avocado/docs/source/api/optional-plugins/avocado_runner_vm/avocado_runner_vm.rst:7: WARNING: autodoc: failed to import module u'avocado_runner_vm'; the following exception was raised:
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/sphinx/ext/autodoc.py", line 551, in import_object
    __import__(self.modname)                                                                                                                                                
ImportError: No module named avocado_runner_vm
```

The reason is that the modules are not visible and can not be imported.  A `make develop` solves that, but the problem is that a `make develop` (or similar) is not done on rtd, leading to an also empty content:

http://avocado-framework-ldoktor.readthedocs.io/en/latest/api/optional-plugins/avocado_varianter_yaml_to_mux/avocado_varianter_yaml_to_mux.html